### ISSUE TITLE
Speed in ddecal

### ISSUE THREAD

**Issue by revoltek** [lofar-astron/DP3#102]
The same diagonal solve in DIE solver and DDE solver (using MODEL_DATA and saving solution in h5parm) take a very different amount of time:

DDEcal: Total NDPPP time    1508.08 real     1508.19 user        3.75 system
DIEcal: Total NDPPP time     107.88 real       94.47 user        3.69 system

is there a fundamental reason why the same solve is so much slower with the new solver?

**Comment by aroffringa** [comment]
Yes, DDECal is slower because it takes a more abstract approach. Moreover, it doesn't have a diagonal solve, so it's doing a constrained full-Jones solve to come up with a diagonal solution. The latter could be fixed, but DDECal will always be slower because of its more generic approach (and thus allowing solving in multiple directions). 

That said, apart from the already mentioned speed-up in diagonal solutions, more efficiency could be gained by further optimization of the DDE algorithm. However, we don't have resources for that.
